### DISCUSSION TITLE
Migrate WPFHost

### DISCUSSION THREAD

**Discussion by glcjr** [unoplatform/uno#16616] [General]
### Your feedback

I decided to checkout Uno. I usually do Windows Desktop and .net Framework so this is probably the third or so time I messed around with Core so this may be all second nature. But...

I started a .Net 6 UWP (so I'd have a designer and a WPF project(it didn't dawn on me it would be a Core as well)) project and VS immediately updated the project to .net 8 and of course results in WPF issues. IE I ran the WPF project and it said I need to migrate the WPF Host. So I followed the steps and there was a bunch of errors resulted.

On a lark I readded the MainWindow.Xaml and cs files. And this resulted in the app running and I see it behind a WPF window telling me I need to migrate again. So I made the WPF's mainwindow visibility to hidden and the project runs as desired. Except in debug mode, a command window opens behind the main window and says something about three controls: CommandFlyout, TextFlyout, and scrollviewer not being implemented, But the release version doesn't do that.

Since I came across a bunch of people having issues with the WPF when I was searching for solutions to the big long list of error thing, I thought that might be helpful. It also might set off red flags to someone that actually knows something about net CORE since I don't.

**Comment by jeromelaban** [comment]
Thanks for the feedback. A few things:
- You're saying you run into WPF issues when updating to .NET 8, could you tell more about those?
- You're mentioning hiding the WPF mainwindow, what's the use here? Do you have other code that creates additional windows?
- You're also mentioning "migrating again"? Can you tell more about what prompted to do that?
- `CommandFlyout`, `TextFlyout`, and `scrollviewer` warnings are about some styling that we're working on that are referencing pieces from the WinUI that are not yet implemented in Uno.

Thanks!

**Reply by glcjr** [reply]
Well I just discovered it was because of user stupidity. IE mine. I messed up the first project I started because Instead of using my project name.App(), I copied the code with AppHead and it didn't recognize AppHead. I blamed that on VS automatically turning it into .net 8 instead of 6.

So I messed the project up bad trying to fix that and then deleted the project and created a new one (since it was simply the helloworld project created by the template) after I installed .net 6 targetting because I thought that was why VS upgraded the project to 8 but it did it again. 

Well with the second project I figured out it was ProjectName.App() instead of AppHead but I forgot to update the Xaml with the new project. It still had the StartupUri="Wpf/MainWindow.xaml" in it. Hence why it still was loading that window and erroring when I started it without the files... Oops.

**Reply by jeromelaban** [reply]
Thanks for the update. Out of curiosity, you chose UWP specifically for the designer, right? Have you tried XAML hot reload? This would allow to use the default uno app template which is easier to manage than the older Uno UWP template.
